migrate/live-migration/live-migration-from-timescaledb.md

```diff
@@ -0,0 +1,159 @@
+your production applications, using the [live migration] strategy. To simplify
+the migration, we provide you with a docker image containing all the tools and
+scripts that you need to perform the live migration.
+
+You should provision a dedicated instance to run the migration steps from.
+Ideally an AWS EC2 instance that's in the same region as the Timescale target service.
+For an ingestion load of 10,000 transactions/s, and assuming that the historical
+data copy takes 2 days, we recommend 4 CPUs with 4 to 8 GiB of RAM and 1.2 TiB of storage.
+
+<SourceTargetNote />
+
+In detail, the migration process consists of the following steps:
+1. Set up a target database instance in Timescale.
+1. Prepare the source database for live migration.
+1. Run the live migration docker image.
+1. Validate the data in target database and use it as new primary.
+
+<GettingHelp />
+
+<StepOne />
+
+## 2. Prepare the source database for live migration
+
+<LiveMigrationStep2 />
+
+Next, you need to ensure that your source tables and hypertables have either a primary key
+or `REPLICA IDENTITY` set. This is important as it is a requirement for replicating `DELETE` and
+`UPDATE` operations. Replica identity assists the replication process in identifying the rows
+being modified. It defaults to using the table's primary key.
+
+If a table doesn't have a primary key, you'll have to manually set the replica identity.
+One option is to use a unique, non-partial, non-deferrable index that includes only columns
+marked as `NOT NULL`. This can be set as the replica identity:
+
+```sh
+ALTER TABLE {table_name} REPLICA IDENTITY USING INDEX {_index_name}
+```
+
+If there's no primary key or viable unique index to use, you have to set `REPLICA IDENTITY`
+to `FULL`. If you are expecting a large number of `UPDATE` or `DELETE` operations on the table,
+using `FULL` is not recommended. For each `UPDATE` or `DELETE` statement, PostgreSQL reads the
+whole table to find all matching rows, resulting in significantly slower replication.
+
+```sh
+ALTER TABLE {table_name} REPLICA IDENTITY FULL
+```
+
+<Highlight type="important">
+TimescaleDB 2.12 and above allows setting `REPLICA IDENTITY` on hypertables. However,
+if you're using a version below 2.12, it’s important that your hypertables have primary keys.
+If they don't, you will need to upgrade your TimescaleDB instance. This is because, without a
+primary key or `REPLICA IDENTITY`, operations such as `DELETE` and `UPDATE` will not be replicated,
+which will impact the consistency of your data post migration.
+</Highlight>
+
+## 3. Run live migration docker image
+
+First, set the database URIs as environment variables:
+
+```sh
+export SOURCE=""
+export TARGET=""
+```
+
+Next, download and run the live migration docker image:
+
+```sh
+docker run --rm -dit --name live-migration \
+  -e PGCOPYDB_SOURCE_PGURI=$SOURCE \
+  -e PGCOPYDB_TARGET_PGURI=$TARGET \
+  -v ~/tmp/live-migration:/opt/timescale/ts_cdc \
```

Comment:
It is better to be explit about the host directory which is being mounted to docker must have enough disk space.

```suggestion
  -v ~/live-migration:/opt/timescale/ts_cdc \
```

Reply:
I have also added a note below this line

```
Ensure `/live-migration` directory has sufficient space available.
Ideally, 1.5 times the source database size should be good.
```